File: Clases 2018/Clases-C#/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/DataLayer/DAOs/PrioridadDao.vb
Public Class PrioridadDao
    Public Function getAll() As List(Of Prioridad)
        Dim lst As New List(Of Prioridad)
        For Each row As DataRow In BDHelper.getDBHelper.ConsultaSQL("Select * from Prioridades").Rows
            lst.Add(map(row))
        Next
        Return lst
    End Function

    Private Function map(ByRef row As DataRow) As Prioridad
        Dim oPrioridad As New Prioridad
        oPrioridad.id_prioridad = Convert.ToInt32(row.Item("id_prioridad").ToString)
        oPrioridad.nombre = row.Item("n_prioridad").ToString
        Return oPrioridad
    End Function
End Class

File: Clases 2018/Clases-C#/ClasePractica11_2018 (update tabla Bugs - asignar bug)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/PrioridadDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class PrioridadDao
    {
        public List<Prioridad> getAll()
        {
            List<Prioridad> lst = new List<Prioridad>();
            foreach (DataRow row in BDHelper.getBDHelper().ConsultaSQL("Select * from Prioridades").Rows)
                lst.Add(map(row));
            return lst;
        }

        private Prioridad map(DataRow row)
        {
            Prioridad oPrioridad = new Prioridad();
            oPrioridad.id_prioridad = Convert.ToInt32(row["id_prioridad"].ToString());
            oPrioridad.nombre = row["n_prioridad"].ToString();
            return oPrioridad;
        }
    }
}
